lib/AST/ASTMangler.cpp

```diff
@@ -2372,6 +2376,20 @@ void ASTMangler::appendImplFunctionType(SILFunctionType *fn,
   appendOperator("I", StringRef(OpArgs.data(), OpArgs.size()));
 }
 
+void ASTMangler::appendElementArchetype(ElementArchetypeType *archetype,
+                                        GenericSignature sig,
+                                        const ValueDecl *forDecl) {
+  ASSERT(forDecl && "Can currently only mangle element archetypes for a Decl");
+  CaptureInfo captureInfo;
+  if (auto fn = AnyFunctionRef::fromDeclContext(forDecl->getDeclContext()))
+    captureInfo = fn->getCaptureInfo();
+
+  auto interfaceTy = mapLocalArchetypesOutOfContext(
+      archetype, sig, captureInfo.getGenericEnvironments());
```

Comment:
Why not just call mapLocalArchetypesOutOfContext() before calling into the mangler?

Reply:
Not sure if this is what you're asking for, but could move this to `getDeclTypeForMangling` instead? I don't really have any strong opinions one way or the other

Reply:
Updated to move into `getDeclTypeForMangling`, how does this look @slavapestov?